src/main/java/seedu/address/logic/commands/AddExamCommand.java

```diff
@@ -0,0 +1,69 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.exam.Exam;
+
+/**
+ * Adds an exam to the address book.
+ */
+public class AddExamCommand extends Command {
+
+    public static final String COMMAND_WORD = "addExam";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds an exam to the address book. "
+            + "Parameters: EXAM_NAME EXAM_DETAILS\n"
+            + "Example: " + COMMAND_WORD + " "
+            + "Math Final Important exam";
+
+    public static final String MESSAGE_SUCCESS = "New exam added: %1$s";
+    public static final String MESSAGE_DUPLICATE_EXAM = "This exam already exists in the address book";
+
+    private final Exam toAdd;
+
+    /**
+     * Creates an AddExamCommand to add the specified {@code Exam}
+     */
+    public AddExamCommand(Exam exam) {
+        requireNonNull(exam);
+        toAdd = exam;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
```

Comment:
Great job on the execute function, it flows logically and has the necessary checks.